presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/SchemaAccessControlRule.java

```diff
@@ -24,26 +24,42 @@
 public class SchemaAccessControlRule
 {
     private final boolean owner;
+    private final Optional<Boolean> visible;
     private final Optional<Pattern> userRegex;
     private final Optional<Pattern> schemaRegex;
 
     @JsonCreator
     public SchemaAccessControlRule(
             @JsonProperty("owner") boolean owner,
+            @JsonProperty("visible") Optional<Boolean> visible,
             @JsonProperty("user") Optional<Pattern> userRegex,
             @JsonProperty("schema") Optional<Pattern> schemaRegex)
     {
         this.owner = owner;
+        this.visible = requireNonNull(visible, "visible is null");
         this.userRegex = requireNonNull(userRegex, "userRegex is null");
         this.schemaRegex = requireNonNull(schemaRegex, "sourceRegex is null");
     }
 
-    public Optional<Boolean> match(String user, String schema)
+    public Optional<Boolean> isSchemaOwner(String user, String schema)
     {
         if (userRegex.map(regex -> regex.matcher(user).matches()).orElse(true) &&
                 schemaRegex.map(regex -> regex.matcher(schema).matches()).orElse(true)) {
             return Optional.of(owner);
         }
         return Optional.empty();
     }
+
+    public Optional<Boolean> isSchemaVisible(String user, String schema)
```

Comment:
I think that method could simply return `boolean` with `true` in case of `Optional.empty()`

Reply:
empty means ` (user, schema) ` does not match, so we should further look for next pattern